include/envoy/ssl/connection.h

```diff
@@ -129,6 +130,15 @@ class ConnectionInfo {
    *         connection.
    **/
   virtual const std::string& tlsVersion() const PURE;
+
+  /**
+   * Retrieves the value of a specific X.509 extension from the peer cert, if a peer cert exists
+   * and it contains the specified extension.
```

Comment:
Please comment that this is used out-of-tree, with a bread-crumb for who to ask about it, so it doesn't get deleted on accident as dead code.